includes/class-freemius.php

```diff
@@ -1432,6 +1452,181 @@ private function _register_hooks() {
+                    if ( false !== strpos( $plugin->package, 'api.freemius' ) ) {
+                        unset( $updates->response[ $file ] );
+                    }
+                }
+            }
+
+            return $updates;
+        }
+
+        /**
+         * The following variables are used to keep track of the plugins that are being handled one at time on the
+         * "Plugins" and "Updates" admin pages in order to determine which plugin details URLs should be modified so
+         * that when the plugin information dialog is shown, it will show data from Freemius if necessary.
+         *
+         * @author Leo Fajardo (@leorw)
+         * @since 2.2.3
+         *
+         * @var array $_plugin_files
+         * @var array $_plugin_fs_instances
+         * @var int   $_plugin_fs_instances_current_index
+         */
+        private static $_plugin_files                      = array();
+        private static $_plugin_fs_instances               = array();
+        private static $_plugin_fs_instances_current_index = -1;
+
+        /**
+         * @author Leo Fajardo (@leorw)
+         * @since 2.2.3
+         */
+        static function _register_plugin_install_urls_hooks() {
+            if ( self::is_updates_page() ) {
+                $plugins = get_plugin_updates();
+            } else {
+                global $wp_list_table;
+                $plugins = $wp_list_table->items;
+            }
+
+            if ( empty( $plugins ) ) {
+                return;
+            }
+
+            $has_any_instance    = false;
+            self::$_plugin_files = array();
+
+            foreach ( $plugins as $plugin_file => $plugin_data ) {
+                if ( self::is_plugins_page() && ! isset( $plugin_data['slug'] ) ) {
+                    // In the core WordPress logic, the details link is shown on the "Plugins" page only when the slug is set.
+                    continue;
+                }
+
+                self::$_plugin_files[]        = $plugin_file;
+                $fs                           = self::get_instance_by_file( $plugin_file );
+                self::$_plugin_fs_instances[] = $fs;
+
+                if ( is_object( $fs ) ) {
+                    $has_any_instance = true;
+                }
+            }
+
+            if ( ! $has_any_instance ) {
+                return;
+            }
+
+            add_filter( 'admin_url', array( 'Freemius', '_prepend_fs_allow_updater_and_dialog_flag_url_param' ), 10, 3 );
+            add_filter( 'network_admin_url', array( 'Freemius', '_prepend_fs_allow_updater_and_dialog_flag_url_param' ), 10, 3 );
+        }
+
+        /**
+         * Prepends the `fs_allow_updater_and_dialog` param to the plugin information URLs to tell the SDK to handle
+         * the information that is shown on the plugin details dialog that is shown when the relevant link is clicked.
+         *
+         * @author Leo Fajardo (@leorw)
+         * @since 2.2.3
+         *

```

Comment:
Add `@global array $wp_current_filter Stores the list of current filters with the current one last.`